### PR TITLE
[r] Fix BPCells ingestion

### PR DESCRIPTION
**Pull request by mojaveazure**
Address issues with BPCells ingestion by re-indexing each chunk's `soma_dim_0` from the span of the chunk matrix to the span of the input matrix. Previously, each chunk would always be written in the span of `soma_dim_0 = [0, stride]`, this needs to be fixed to be the span of the input matrix

Also add round-trip tests to ensure that the SOMA array matches the input matrix

Fixes [SOMA-719](https://linear.app/tiledb/issue/SOMA-719/bpcells-ingestion-produces-incorrect-expression-values)

### CODE REVIEW

File: apis/r/tests/testthat/test-13-write-soma-objects.R [hunk 1]
```diff
@@ -526,7 +528,45 @@ test_that("write_soma.IterableMatrix registration", {
     ))
     expect_error(write_soma(bpmat, "uri", soma_parent = NULL, key = "knex"))
   }
+})
 
+test_that("write_soma.IterableMatrix integrity", {
```
[the remark this answers is not in the document]
Comment (mojaveazure) [reply apis/r/tests/testthat/test-13-write-soma-objects.R]: > Does this test a multi-chunk scenario?

Yes, the size of the matrix was chosen to have two chunks of unequal sizes
```r
> mat <- create_sparse_matrix_with_int_dims(nrows = 1610L, ncols = 560L)
> ctx <- SOMATileDBContext$new(c(soma.init_buffer_bytes = "33554432"))
> (stride <- .block_size(ncol(mat), tiledbsoma_ctx = ctx))
[1] 1069
> nrow(mat) > stride # TRUE
[1] TRUE
> ceiling(nrow(mat) / stride) # number of chunks
[1] 2
```

> It's also worth adding a multi-chunk test that contains an empty chunk to make sure we handle that edge case gracefully

That's a great idea, will add tests for that
